src/serde/ser.rs

```diff
@@ -0,0 +1,621 @@
+use std::io::Cursor;
+
+use crate::ion_writer::IonWriter;
+use crate::result::IonFailure;
+use crate::serde::decimal::TUNNELED_DECIMAL_TYPE_NAME;
+use crate::serde::timestamp::TUNNELED_TIMESTAMP_TYPE_NAME;
+use crate::types::Int;
+use crate::{
+    BinaryWriterBuilder, Decimal, IonError, IonResult, IonType, TextKind, TextWriterBuilder,
+    Timestamp,
+};
+use serde::ser::Impossible;
+use serde::{ser, Serialize};
+
+/// Serialize an object into pretty formatted Ion text
+pub fn to_pretty<T>(value: &T) -> IonResult<String>
```

Comment:
We have a `Format` enum you could take:
```rust
to_ion(Format::Text(TextKind::Pretty))?;
to_ion(Format::Text(TextKind::Lines))?;
to_ion(Format::Text(TextKind::Compact))?;
to_ion(Format::Binary)?;
```
Of course, that doesn't capture the upcoming availability of Ion 1.1...

I think we need a write config builder API that we can use for things like these. See #618.

Reply:
https://github.com/amazon-ion/ion-rust/issues/646